File: sdkwork-backend-sdk-flutter/lib/src/api/ai_audio_generation.dart
import '../http/client.dart';
import '../models.dart';

class AiAudioGenerationApi {
  final HttpClient _client;
  
  AiAudioGenerationApi(this._client);

  /// Create audio generation task
  Future<PlusApiResultGenerateAudioVO?> create(GenerateAudioForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/generation/audio/create'), body: body, contentType: 'application/json');
    return response is PlusApiResultGenerateAudioVO ? response : null;
  }

  /// Get audio generation result
  Future<PlusApiResultGenerateAudioVO?> getResult(String taskId) async {
    final response = await _client.get(ApiPaths.backendPath('/generation/audio/result/${taskId}'));
    return response is PlusApiResultGenerateAudioVO ? response : null;
  }
}

File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/ai_audio_generation.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import GenerateAudioForm, PlusApiResultGenerateAudioVO

class AiAudioGenerationApi:
    """ai_audio_generation API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def create(self, body: GenerateAudioForm) -> PlusApiResultGenerateAudioVO:
        """Create audio generation task"""
        return self._client.post(f"/backend/v3/api/generation/audio/create", json=body)

    def get_result(self, taskId: str) -> PlusApiResultGenerateAudioVO:
        """Get audio generation result"""
        return self._client.get(f"/backend/v3/api/generation/audio/result/{taskId}")
